services/registration/controller/controller.go

```diff
@@ -126,6 +126,10 @@ func CreateCurrentUserRegistration(w http.ResponseWriter, r *http.Request) {
 	err = service.CreateUserRegistration(id, user_registration)
 
 	if err != nil {
+		if err.Error() == service.RegistrationAlreadyExists {
+			errors.WriteError(w, r, errors.ConflictError(err.Error(), "Could not create user registration."))
```

Comment:
I apologize if the control flow is a little messy. I'll fix it after the DB level fix for 404 errors is merged (#417)